### DISCUSSION TITLE
nar hash mismatch for pkgs

### DISCUSSION THREAD

**Discussion by benneti** [nrdxp/nixdev#243] [Q&A]
Hey,

does anyone has a bit more understanding how the NAR hash ist calculated;
If I create the pkgs flake lock and share it with another computer I get a mismatch.
Is there anything I have to ensure, like exactly the same rights particularities about the filesystems?

**Comment by codygman** [comment]
I get those too recently, but only on a computer with an older version of nix. The others seem to agree, so I think it has something to do with an update.

**Reply by benneti** [reply]
I don't think this is the case for me both computers have version nix (Nix) 2.4pre20210326_dd77f71
It could be that the repos have different rights somewhere (but at least on the top level everything matches)

**Comment by nrdxp** [comment]
I think I need to fix the `flk update` command again. I think `nix flake lock` is more approprate after recent updates. This _may_ resolve your issues. For now, just try `nix flake lock --update-input pkgs` manually, and see if that helps.

**Reply by benneti** [reply]
Ah thanks thats good to know, indeed I am using nix flake metadata to update. I will try and report back

**Reply by benneti** [reply]
while this makes --update-input nicer again there is still a mismatch on the different PCs

**Reply by codygman** [reply]
> while this makes --update-input nicer again there is still a mismatch on the different PCs

Yeah, I'm being affected by this now. It seems concerning there's different output on different machines.

**Reply by benneti** [reply]
It seems like it fixed itself with a recent update (sorry I did not bysect which nixpkgs version exactly).
